### PR TITLE
dialects: Adding integer width arith ops

### PR DESCRIPTION
**Pull request by ajrichins**
adding arith operations TruncIOp, ExtSIOp and ExtUIOp.

@Hatsunespica

### CODE REVIEW

File: tests/dialects/test_arith.py [hunk 1]
```diff
@@ -245,3 +248,18 @@ def test_cmpi_index_type():
...
     Cmpi.get(a, b, "eq").verify()
+
+
+def test_extend_truncate_iops():
+    a = Constant.from_int_and_width(1, i32)
+    b = Constant.from_int_and_width(2, i64)
+    exts_op = ExtSIOp.get(a, i64)
+    extu_op = ExtUIOp.get(a, i64)
+    trunc_op = TruncIOp.get(b, i32)
+
```
Comment (math-fehr) [inline tests/dialects/test_arith.py:259]: Could you just add `exts_op.verify()`, and same for the two others?

**Review by math-fehr — APPROVED** [review]
Perfect, thanks a lot!
I just added a comment on the test, but otherwise that looks good to me!

File: xdsl/dialects/arith.py [hunk 2]
```diff
@@ -619,6 +619,42 @@ def get(op: SSAValue | Operation, target_type: AnyFloat):
...
 
+@irdl_op_definition
+class TruncIOp(IRDLOperation):
+    name = "arith.trunci"
+
+    input: Operand = operand_def(IntegerType)
+    result: OpResult = result_def(IntegerType)
+
+    @staticmethod
+    def get(op: SSAValue | Operation, target_type: IntegerType):
```
Comment (superlopuh) [inline xdsl/dialects/arith.py:630]: Can you please use inits? We haven't migrated every file yet but I feel that new operations should use initialisers

Comment (math-fehr) [reply xdsl/dialects/arith.py:630]: Oh yes! The only change would be to have `def __init__(self, ...)` instead of `get` and the `staticmethod`!

**Review by math-fehr — APPROVED** [review]
Perfect, thanks a lot!

**Pull request merged**